lambdas/es/indexer/test/test_index.py

```diff
@@ -19,130 +22,216 @@
-                "eTag": "123456",
-                "sequencer": "0A1B2C3D4E5F678901",
-                "versionId": "1313131313131.Vier50HdNbi7ZirO65"
-            },
-            "s3SchemaVersion": "1.0"
+        "configurationId": "testConfigRule",
+        "object": {
+            "key": "hello+world.txt",
+            "sequencer": "0A1B2C3D4E5F678901"
         },
-        "userIdentity": {
-            "principalId": "EXAMPLE"
-        }
-    }
+        "s3SchemaVersion": "1.0"
+    },
+    "userIdentity": {"principalId": "EXAMPLE"}
 }
 
 
+def _check_event(synthetic, organic):
+    # Ensure that synthetic events have the same shape as actual organic ones,
+    # and that overridden properties like bucket, key, eTag are properly set
+    # same keys at top level
+    assert organic.keys() == synthetic.keys()
+    # same value types (values might differ and that's OK)
+    assert {type(v) for v in organic.values()} == \
+        {type(v) for v in synthetic.values()}
+    # same keys and nested under "s3"
+    assert organic["s3"].keys() == synthetic["s3"].keys()
+    assert organic["s3"]["bucket"].keys() == synthetic["s3"]["bucket"].keys()
+    # same value types under S3 (values might differ and that's OK)
+    assert {type(v) for v in organic["s3"].values()} == \
+        {type(v) for v in synthetic["s3"].values()}
+    # spot checks for overridden properties
+    # size absent on delete
+    if "size" in organic["s3"]["bucket"]:
+        assert organic["s3"]["object"]["size"] == synthetic["s3"]["object"]["size"]
+    # versionId absent when unversioned bucket or hard delete
+    if "versionId" in organic["s3"]["bucket"]:
+        assert organic["s3"]["object"]["versionId"] == synthetic["s3"]["object"]["versionId"]
+    # should always be present
+    assert organic["awsRegion"] == synthetic["awsRegion"]
+    assert organic["s3"]["bucket"]["arn"] == synthetic["s3"]["bucket"]["arn"]
+    assert organic["s3"]["bucket"]["name"] == synthetic["s3"]["bucket"]["name"]
+    assert organic["s3"]["object"]["key"] == synthetic["s3"]["object"]["key"]
+    assert organic["s3"]["object"]["eTag"] == synthetic["s3"]["object"]["eTag"]
+
+
+def _make_es_callback():
+    """
+    create a callback that checks the shape of the response
+    """
+    def check_response(request):
+        raw = [json.loads(line) for line in request.body.splitlines()]
+        # drop the optional source and isolate the actions
+        # see https://www.elastic.co/guide/en/elasticsearch/reference/6.7/docs-bulk.html
+        actions = [deepcopy(l) for l in raw if len(l.keys()) == 1]
+        def to_response(d):
+            top_key = next(iter(d.keys()))
+            values = d[top_key]
+            return {
+                top_key: {
+                    "_id": values["_id"],
+                    "_index": values["_index"],
+                    "_type": "_doc",
+                    "status": 200
+                }
+            }
+        items = list(map(to_response, actions))
```

Comment:
```suggestion
        actions = [deepcopy(l) for l in raw if len(l) == 1]
        items = [
            {
                top_key: {
                    "_id": values["_id"],
                    "_index": values["_index"],
                    "_type": "_doc",
                    "status": 200
                }
            }
            for action in actions
            for top_key, values in action.items()
        ]
```
Actually I don't quite understand why `deepcopy()` is needed here.

Reply:
`deepcopy` is no longer needed as we no longer copy deep values; removed